src/sage/algebras/quatalg/quaternion_algebra.py

```
@@ -2870,30 +2938,142 @@ def is_equivalent(self, J, B=10) -> bool:
+            (True, 1/810*i + 1/405*k)
+        """
+        if side == 'left':
+            return self.is_left_equivalent(J, B, certificate)
+        # If None, assume right ideals, for backwards compatibility
+        return self.is_right_equivalent(J, B, certificate)
+
+    def is_left_equivalent(self, J, B=10, certificate=False):
+        r"""
+        Checks whether ``self`` and ``J`` are equivalent as left ideals.
+        Requires the underlying quaternion algebra to be definite.
+
+        INPUT:
+
+        - ``J`` -- a fractional quaternion left ideal with same order as ``self``
+
+        - ``B`` -- a bound to compute and compare theta series before
+          doing the full equivalence test
+
+        - ``certificate`` -- if ``True`` returns an element alpha such that J*alpha=I
+
+        OUTPUT: bool, or (bool, alpha) if ``certificate`` is ``True``
+        """
+        if certificate:
+            is_equiv, cert = self.conjugate().is_right_equivalent(J.conjugate(), B, True)
+            if is_equiv:
+                return True, cert.conjugate()
+            return False, None
+        return self.conjugate().is_right_equivalent(J.conjugate(), B, False)
+
+    def is_right_equivalent(self, J, B=10, certificate=False):
+        r"""
+        Checks whether ``self`` and ``J`` are equivalent as right ideals.
+        Requires the underlying quaternion algebra to be definite.
+
+        INPUT:
+
+        - ``J`` -- a fractional quaternion right ideal with same order as ``self``
+
+        - ``B`` -- a bound to compute and compare theta series before
+          doing the full equivalence test
+
+        - ``certificate`` -- if ``True`` returns an element alpha such that alpha*J=I
+
+        OUTPUT: bool, or (bool, alpha) if ``certificate`` is ``True``
         """
-        # shorthand: let I be self
-        if not isinstance(self, QuaternionFractionalIdeal_rational):
+        if not isinstance(J, QuaternionFractionalIdeal_rational):
```

I'd say in this case we should probably throw a `TypeError`. Returning `False` is arguably correct, but I can't see any situation in which users will rely on this behaviour wilfully, hence failing visibly is better.